### PR TITLE
Return `NaN` rather than throw if provided invalid numeric values

### PR DESCRIPTION
**Pull request by Planeshifter**
This pull request changes base special functions from throwing errors to consistently returning `NaN`.

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/erfinv/test/test.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ tape( 'if provided `0`, the function returns `0`', function test( t ) {
 tape( 'if provided a value which is either less than `-1` or greater than `1`, the function throws a range error', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/erfinv/test/test.js]: @Planeshifter Test description should be updated.

File: lib/node_modules/@stdlib/math/base/special/acos/test/test.js [hunk 2]
```diff
@@ -108,29 +108,17 @@ tape( 'the function throws an error if provided a value less than `-1`', functio
...
 	t.end();
-
-	function badValue( value ) {
-		return function badValue() {
-			acos( value );
-		};
-	}
 });
 
 tape( 'the function throws an error if provided a value greater than `+1`', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acos/test/test.js]: @Planeshifter Test description needs to be updated.

File: lib/node_modules/@stdlib/math/base/special/acos/test/test.js [hunk 3]
```diff
@@ -108,29 +108,17 @@ tape( 'the function throws an error if provided a value less than `-1`', functio
 	var i;
 	for ( i = 0; i < 1e4; i++ ) {
 		v = -randu()*1e6 - (1+EPS);
-		t.throws( badValue( v ), RangeError, 'throws a range error when provided '+v );
+		t.equal( isnan( acos( v ) ), true, 'returns NaN when provided '+v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acos/test/test.js:111]: @Planeshifter Test description needs updated.

File: lib/node_modules/@stdlib/math/base/special/asin/test/test.js [hunk 4]
```diff
@@ -103,21 +103,15 @@ tape( 'the function returns `NaN` if provided a `NaN`', function test( t ) {
...
 	t.end();
-
-	function badValue( value ) {
-		return function badValue() {
-			asin( value );
-		};
-	}
 });
 
 tape( 'the function throws a range error if provided a value greater than `+1`', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/test/test.js]: @Planeshifter Test description needs updated.

File: lib/node_modules/@stdlib/math/base/special/atanh/test/test.js [hunk 5]
```diff
@@ -119,32 +119,20 @@ tape( 'the function throws a range error if provided a value less than `-1`', fu
 
 	for ( i = 0; i < 1e3; i++ ) {
 		v = -randu()*1e6 - (1.0+EPS);
-		t.throws( badValue( v ), RangeError, 'throws a range error when provided '+v );
+		t.equal( isnan( atanh( v ) ), true, 'returns NaN when provided '+v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/atanh/test/test.js:122]: @Planeshifter Test description needs updated.

**Comment by Planeshifter** [comment]
Fixed, thanks!

**Pull request merged**